File: src/spec/resources/testprojects/nodetest/api/store-api.ts
import type { ApiClient } from '../api-client.js';
import { BaseApi } from './base-api.js';
import { Configuration } from '../configuration.js';
import { ObjectSerializer } from '../object-serializer.js';
import { Order } from '../models/index.js';

/**
 * StoreApi provides methods for the Store API group.
 */
export class StoreApi extends BaseApi {
  constructor(config?: Configuration, apiClient?: ApiClient) {
    super(config, apiClient);
  }

  /**
   * Delete purchase order by ID
   * @param orderId ID of the order to delete (required)
   */
  async deleteOrder(orderId: number): Promise<void> {
    if (orderId == null) {
      throw new Error('Missing required parameter "orderId" when calling deleteOrder');
    }
    let path = `/store/order/{orderId}`;
    path = path.replace(`{${'orderId'}}`, encodeURIComponent(ObjectSerializer.toPathValue(orderId)));
    const queryParams: Record<string, unknown> = {};
    const headerParams: Record<string, string> = {};
    (await this.invokeApi('DELETE', path, queryParams, headerParams, null, [], 'application/json', null)) as void;
  }

  /**
   * Returns pet inventories by status
   * @return { [key: string]: number }
   */
  async getInventory(): Promise<{ [key: string]: number }> {
    const path = `/store/inventory`;
    const queryParams: Record<string, unknown> = {};
    const headerParams: Record<string, string> = {};
    return (await this.invokeApi(
      'GET',
      path,
      queryParams,
      headerParams,
      null,
      ['application/json'],
      'application/json',
      (json: unknown) => json as { [key: string]: number }
    )) as { [key: string]: number };
  }

  /**
   * Find purchase order by ID
   * @param orderId ID of order to return (required)
   * @return Order
   */
  async getOrderById(orderId: number): Promise<Order> {
    if (orderId == null) {
      throw new Error('Missing required parameter "orderId" when calling getOrderById');
    }
    let path = `/store/order/{orderId}`;
    path = path.replace(`{${'orderId'}}`, encodeURIComponent(ObjectSerializer.toPathValue(orderId)));
    const queryParams: Record<string, unknown> = {};
    const headerParams: Record<string, string> = {};
    return (await this.invokeApi(
      'GET',
      path,
      queryParams,
      headerParams,
      null,
      ['application/json'],
      'application/json',
      (json: unknown) => ObjectSerializer.deserialize(json, Order)
    )) as Order;
  }

  /**
   * Place an order for a pet
   * @param order  (optional)
   * @return Order
   */
  async placeOrder(order?: Order): Promise<Order> {
    const path = `/store/order`;
    const queryParams: Record<string, unknown> = {};
    const headerParams: Record<string, string> = {};
    return (await this.invokeApi(
      'POST',
      path,
      queryParams,
      headerParams,
      order,
      ['application/json'],
      'application/json',
      (json: unknown) => ObjectSerializer.deserialize(json, Order)
    )) as Order;
  }
}

File: src/spec/resources/testprojects/pytest/petstore_client/api/store_api.py
from typing import Any, Dict, List, Optional, Union
from urllib.parse import quote

from petstore_client.models.order import Order

from ..api_client import ApiClient
from ..default_api_client import DefaultApiClient
from ..configuration import Configuration
from .base_api import BaseApi
from ..object_serializer import ObjectSerializer


class StoreApi(BaseApi):
    """StoreApi provides methods for the store API group."""

    def __init__(
        self,
        api_client: Optional[ApiClient] = None,
        config: Optional[Configuration] = None,
    ):
        super().__init__(api_client, config)

    def delete_order(
        self,
        order_id: int,
    ) -> None:
        """Delete purchase order by ID
        :param order_id: ID of the order to delete (required)
        """
        if order_id is None:
            raise ValueError("Missing the required parameter 'order_id'")

        path = '/store/order/{orderId}'
        path = path.replace('{' + 'orderId' + '}', quote(ObjectSerializer.to_path_value(order_id), safe=''))
        query_params: Dict[str, Any] = {}
        header_params: Dict[str, str] = {}
        body = None

        return self._invoke_api(  # type: ignore[no-any-return]
            'DELETE',
            path,
            query_params,
            header_params,
            body,
            [],
            'application/json',
            None,
            None,
        )

    def get_inventory(
        self,
    ) -> Dict[str, int]:
        """Returns pet inventories by status
        :return: Dict[str, int]
        """
        path = '/store/inventory'
        query_params: Dict[str, Any] = {}
        header_params: Dict[str, str] = {}
        body = None

        return self._invoke_api(  # type: ignore[no-any-return]
            'GET',
            path,
            query_params,
            header_params,
            body,
            ['application/json'],
            'application/json',
            'Dict[str, int]',
            None,
        )

    def get_order_by_id(
        self,
        order_id: int,
    ) -> Order:
        """Find purchase order by ID
        :param order_id: ID of order to return (required)
        :return: Order
        """
        if order_id is None:
            raise ValueError("Missing the required parameter 'order_id'")

        path = '/store/order/{orderId}'
        path = path.replace('{' + 'orderId' + '}', quote(ObjectSerializer.to_path_value(order_id), safe=''))
        query_params: Dict[str, Any] = {}
        header_params: Dict[str, str] = {}
        body = None

        return self._invoke_api(  # type: ignore[no-any-return]
            'GET',
            path,
            query_params,
            header_params,
            body,
            ['application/json'],
            'application/json',
            'Order',
            None,
        )

    def place_order(
        self,
        order: Optional[Order] = None,
    ) -> Order:
        """Place an order for a pet
        :param order:  (optional)
        :return: Order
        """
        path = '/store/order'
        query_params: Dict[str, Any] = {}
        header_params: Dict[str, str] = {}
        body = order

        return self._invoke_api(  # type: ignore[no-any-return]
            'POST',
            path,
            query_params,
            header_params,
            body,
            ['application/json'],
            'application/json',
            'Order',
            None,
        )
